homeassistant/components/media_player/broadlink.py

```diff
@@ -0,0 +1,234 @@
+    SUPPORT_PLAY_MEDIA, SUPPORT_PREVIOUS_TRACK, SUPPORT_SELECT_SOURCE,
+    SUPPORT_TURN_OFF, SUPPORT_TURN_ON, SUPPORT_VOLUME_MUTE,
+    SUPPORT_VOLUME_STEP, MediaPlayerDevice)
+from homeassistant.const import (
+    CONF_COMMAND_OFF, CONF_COMMAND_ON, CONF_HOST, CONF_MAC, CONF_NAME,
+    CONF_PORT, CONF_TIMEOUT, STATE_OFF, STATE_ON)
+from homeassistant.exceptions import PlatformNotReady
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['broadlink==0.9.0']
+DOMAIN = 'broadlink'
+
+DEFAULT_NAME = "Broadlink IR Media Player"
+DEFAULT_TIMEOUT = 10
+DEFAULT_DELAY = 0.5
+DEFAULT_PORT = 80
+
+CONF_VOLUME_UP = 'volume_up'
+CONF_VOLUME_DOWN = 'volume_down'
+CONF_VOLUME_MUTE = 'volume_mute'
+CONF_NEXT_TRACK = 'next_track'
+CONF_PREVIOUS_TRACK = 'previous_track'
+CONF_SOURCES = 'sources'
+CONF_CHANNELS = 'channels'
+CONF_DIGITS = 'digits'
+CONF_DIGITDELAY = 'digitdelay'
+
+DIGITS_SCHEMA = vol.Schema({
+    vol.Required('0'): cv.string,
+    vol.Required('1'): cv.string,
+    vol.Required('2'): cv.string,
+    vol.Required('3'): cv.string,
+    vol.Required('4'): cv.string,
+    vol.Required('5'): cv.string,
+    vol.Required('6'): cv.string,
+    vol.Required('7'): cv.string,
+    vol.Required('8'): cv.string,
+    vol.Required('9'): cv.string,
+})
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST): cv.ipv4_address,
+    vol.Required(CONF_PORT, default=DEFAULT_PORT): cv.positive_int,
+    vol.Required(CONF_MAC): cv.mac_address,
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Optional(CONF_TIMEOUT, default=DEFAULT_TIMEOUT): cv.positive_int,
+
+    vol.Optional(CONF_DIGITDELAY, default=DEFAULT_DELAY): float,
+    vol.Optional(CONF_COMMAND_ON): cv.string,
+    vol.Optional(CONF_COMMAND_OFF): cv.string,
+    vol.Optional(CONF_VOLUME_UP): cv.string,
+    vol.Optional(CONF_VOLUME_DOWN): cv.string,
+    vol.Optional(CONF_VOLUME_MUTE): cv.string,
+    vol.Optional(CONF_NEXT_TRACK): cv.string,
+    vol.Optional(CONF_PREVIOUS_TRACK): cv.string,
+    vol.Optional(CONF_SOURCES, default={}): dict,
+    vol.Optional(CONF_DIGITS): DIGITS_SCHEMA,
+})
+
+SUPPORT_MAPPING = [
+    (CONF_COMMAND_ON, SUPPORT_TURN_ON),
+    (CONF_COMMAND_OFF, SUPPORT_TURN_OFF),
+    (CONF_VOLUME_UP, SUPPORT_VOLUME_STEP),
+    (CONF_VOLUME_DOWN, SUPPORT_VOLUME_STEP),
+    (CONF_VOLUME_MUTE, SUPPORT_VOLUME_MUTE),
+    (CONF_NEXT_TRACK, SUPPORT_NEXT_TRACK),
+    (CONF_PREVIOUS_TRACK, SUPPORT_PREVIOUS_TRACK),
+]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_platform(hass,
+                               config,
+                               async_add_devices,
+                               discovery_info=None):
+    """Set up platform."""
+    import broadlink
+
+    host = (config.get(CONF_HOST),
+            config.get(CONF_PORT))
+    mac = get_broadlink_mac(config.get(CONF_MAC))
+
+    link = broadlink.rm(
+        host,
+        mac,
+        None)
```

Comment:
Think its safe to keep everything in one line ;)

Reply:
I'm a stickler for alignment. Check how things where aligned before the hound started messing with my mojo.

Reply:
Hound can be a PITA, but it keeps the project code base sane :)